packit_service/worker/events/event.py

```diff
@@ -320,14 +319,25 @@ def get_non_serializable_attributes(self):
             "_package_config",
         ]
 
-    def get_dict(self, default_dict: Optional[Dict] = None) -> dict:
+    def get_dict(
+        self, default_dict: Optional[Dict] = None, store_project_event=False
+    ) -> dict:
         d = default_dict or self.__dict__
         # whole dict has to be JSON serializable because of redis
         d = self.make_serializable(d, self.get_non_serializable_attributes())
         d["event_type"] = self.__class__.__name__
 
-        # we are trying to be lazy => don't touch database if it is not needed
-        d["event_id"] = self._db_project_object.id if self._db_project_object else None
+        if store_project_event:
+            package_config_dict = self.packages_config.get_raw_dict_with_defaults()
+            db_project_object = self.db_project_object
+            if self.db_project_event and package_config_dict:
```

Comment:
Just not sure what could happen here if `package_config_dict` is None

Reply:
the package config will not be stored to DB, which shouldn't be a problem as the "old" solution (getting the config via API) is still a fallback for the result events